onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -48,7 +48,7 @@ std::string GetEnginePath(const ::std::string& root, const std::string& name) {
 
 std::string GetVecHash(const ::std::vector<int>& vec) {
   std::size_t ret = vec.size();
-  for (auto& i : vec) {
+  for (const auto i : vec) {
```

Comment:
const is not needed if we're doing a copy of primitive type?

Reply:
True, yet this is kind of convention to emphasize the loop variable is a _source_ only (in general, each time I use _source_ in the code, even if its a primitive type, I declare it as a `const`)
WDYT ?

Reply:
the original code should have been
const auto& i : vec if we were going to use references.
the const would protect the reference from being modified.
my understanding is that is the standard use of const in that pattern.

if we are going to change to copy, because it's more efficient for primitive type,
it's probably better to leave off the const (to avoid confusing with the aforementioned pattern for references)

Reply:
Sure will change (just see this https://stackoverflow.com/questions/16033751/c-const-modifier-with-primitive-types)